src/kbmod/data_interface.py

```diff
@@ -3,12 +3,111 @@
 from astropy.io import fits
 from astropy.wcs import WCS
 import numpy as np
-
-import kbmod.search as kb
+from pathlib import Path
 
 from kbmod.configuration import SearchConfiguration
 from kbmod.file_utils import *
-from kbmod.work_unit import WorkUnit
+from kbmod.search import (
+    ImageStack,
+    LayeredImage,
+    PSF,
+    RawImage,
+)
+from kbmod.wcs_utils import append_wcs_to_hdu_header
+from kbmod.work_unit import WorkUnit, raw_image_to_hdu
+
+
+def load_deccam_layered_image(filename, psf):
+    """Load a layered image from the legacy deccam format.
+
+    Parameters
+    ----------
+    filename : `str`
+        The name of the file to load.
+    psf : `PSF`
+        The PSF to use for the image.
+
+    Returns
+    -------
+    img : `LayeredImage`
+        The loaded image.
+
+    Raises
+    ------
+    Raises a ``FileNotFoundError`` if the file does not exist.
+    Raises a ``ValueError`` if any of the validation checks fail.
+    """
+    if not Path(filename).is_file():
+        raise FileNotFoundError(f"{filename} not found")
+
+    img = None
+    with fits.open(filename) as hdul:
+        if len(hdul) < 4:
+            raise ValueError("Not enough extensions for legacy deccam format")
+
+        # Extract the obstime trying from a few keys and a few extensions.
+        obstime = 0.0
+        if "MJD" in hdul[0].header:
+            obstime = hdul[0].header["MJD"]
+        elif "MJD" in hdul[1].header:
+            obstime = hdul[1].header["MJD"]
+
+        img = LayeredImage(
+            RawImage(hdul[1].data.astype(np.float32), obstime),  # Science
+            RawImage(hdul[3].data.astype(np.float32), obstime),  # Variance
+            RawImage(hdul[2].data.astype(np.float32), obstime),  # Mask
+            psf,
+        )
+
+    return img
+
+
+def save_deccam_layered_image(img, filename, wcs=None, overwrite=True):
+    """Save a layered image to the legacy deccam format.
+
+    Parameters
+    ----------
+    img : `LayeredImage`
+        The image to save.
+    filename : `str`
+        The name of the file to save.
+    wcs : `astropy.wcs.WCS`, optional
+        The WCS of the image. If provided appends this information to the header.
+    overwrite : `bool`
+        Indicates whether to overwrite the current file if it exists.
+
+    Raises
+    ------
+    Raises a ``ValueError`` if the file exists and ``overwrite`` is ``False``.
+    """
+    if Path(filename).is_file() and not overwrite:
+        raise ValueError(f"{filename} already exists")
+
+    hdul = fits.HDUList()
+
+    # Create the primary header.
+    pri = fits.PrimaryHDU()
+    pri.header["MJD"] = img.get_obstime()
+    if wcs is not None:
+        append_wcs_to_hdu_header(wcs, pri.header)
```

Comment:
Should all valid deccam images have a WCS? Might make sense to fail if this is missing

Reply:
I don't think they do for all the old data we have used historically. We will want to enforce a WCS in the future, but not in the legacy interface.